frontend/ui/widget/filechooser.lua

```diff
@@ -235,16 +235,18 @@ local FileChooser = Menu:extend{
                     local doc_settings = DocSettings:open(item.path)
                     local summary = doc_settings:readSetting("summary")
 
-                    -- books marked as "finished" should be considered the same as 100%
+                    -- books marked as "finished" or "on hold" should be considered the same as 100% and 0% respectively
                     if summary and summary.status == "complete" then
-                        item.percent_finished = 1.0
-                        return
+                        item.sort_percent = 1.0
+                    elseif summary and summary.status == "abandoned" then
+                        item.sort_percent = 0
                     end
 
                     percent_finished = doc_settings:readSetting("percent_finished")
                 end
                 -- smooth 2 decimal points (0.00) instead of 16 decimal numbers
-                item.percent_finished = math.floor((percent_finished or -1) * 100) / 100
+                item.sort_percent = item.sort_percent or math.floor((percent_finished or -1) * 100) / 100
```

Comment:
What is the purpose of this calculations? Not just
`item.sort_percent = sort_percent or percent_finished or -1`

Reply:
> What is the purpose of these calculations? Not just

no changes done to the calculations in this PR. the original purpose (which is mentioned in #11369) was `smooth percentage to 2 decimal points instead of the numbers initial 16 decimal points (lua didnt provide a function for this to my knowledge)` or for short to sort books which are (for example) `0.000000004` and `0.000003` the same as `0.00` (which is a case when a book is opened, but still on the first page)

Reply:
> (for example) `0.000000004` and `0.000003` the same as `0.00` (which is a case when a book is opened, but still on the first page)

Got it. So, these books will be sorted by name.
In other two "by percent" collate types they will be sorted by original percent.
Maybe we should fix the inconsistency.
As we display two digits only, we can make "smoothing" immediately after
`percent_finished = doc_settings:readSetting("percent_finished")`
for all collate types.